x-pack/test/apm_api_integration/tests/service_nodes/get_service_nodes.spec.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import url from 'url';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import archives from '../../common/fixtures/es_archiver/archives_metadata';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+  const supertest = getService('legacySupertestAsApmReadUser');
+
+  const archiveName = 'apm_8.0.0';
+  const { start, end } = archives[archiveName];
+
+  registry.when(
+    'Service nodes when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      it('handles the empty state', async () => {
+        const response = await supertest.get(
+          url.format({
+            pathname: `/internal/apm/services/opbeans-java/serviceNodes`,
+            query: { start, end, kuery: '', environment: 'ENVIRONMENT_ALL'},
+          })
```

Comment:
You can see an example of how `apmApiClient` is used here: https://github.com/elastic/kibana/blob/main/x-pack/test/apm_api_integration/tests/dependencies/metadata.spec.ts#L20-L28


```suggestion
        const response = await apmApiClient.readUser({
          endpoint: 'GET /internal/apm/services/{serviceName}/serviceNodes'
          params: {
            path: { serviceName: 'opbeans-java'},
            query: { start, end, kuery: '', environment: 'ENVIRONMENT_ALL'},
          })
```